examples/cifar10/train.cpp

```diff
@@ -104,21 +116,75 @@ void train_cifar10(string data_dir_path, double learning_rate, ostream &log) {
                           n_train_epochs, on_enumerate_minibatch,
                           on_enumerate_epoch);
 
-  cout << "end training." << endl;
+  std::cout << "end training." << std::endl;
 
   // test and show results
-  nn.test(test_images, test_labels).print_detail(cout);
-
+  nn.test(test_images, test_labels).print_detail(std::cout);
   // save networks
-  ofstream ofs("cifar-weights");
+  std::ofstream ofs("cifar-weights");
   ofs << nn;
 }
 
+static core::backend_t parse_backend_name(const char *name) {
+  static const char *names[] = {
```

Comment:
Also note that we are not optimizing performance here, this is just an example for standard users which probably will use plane c++11 and since the library itself it's targeted in that, I think it's a good idea to have high level interfaces without mixing low level C.